app/scripts/services/applicationGenerator.js

```diff
@@ -126,12 +126,18 @@ angular.module("openshiftConsole")
           to: {
             kind: "Service",
             name: serviceName
-          }
+          },
+          wildcardPolicy: input.routing.wildcardPolicy
         }
       };
 
-      if (input.routing.host) {
-        route.spec.host = input.routing.host;
+      var host = input.routing.host;
+      if (route.spec.wildcardPolicy === 'Subdomain') {
+        host = 'www.' + input.routing.subdomain;
```

Comment:
I think you need to let them set it, but then let them set the wildcard flag separately.

I realized that we will expose those names when a router doesn't support wildcards, so we really do have to expose the full names to the end user :-(